xl-v1-object.go

```diff
@@ -173,8 +169,17 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64, length i
 			readSize = length - totalBytesRead
 		}
 
+		// Get the checksums of the current part.
+		checkSums := make([]string, len(onlineDisks))
+		for index := range onlineDisks {
+			checkSums[index], _, err = metaArr[index].GetCheckSum(partName)
```

Comment:
change the function signature to not return an error instead rely on the returned values. Since there is only one error being returned 'errUnexpected' it is better handled here. 

```
checkSums[index], _ = metaAttr[index].GetCheckSum(partName) 
if checkSums[index] == "" {
     return toObjectError(errUnexpected, bucket, object)
}
```